solr/src/main/java/com/yahoo/ycsb/db/SolrClient.java

```diff
@@ -0,0 +1,316 @@
+ * </ul>
+ *
+ * @author Gerard Haughian
+ *
+ */
+public class SolrClient extends DB {
+
+  public static final String DEFAULT_COLLECTION_NAME = "ycsb";
+  public static final String DEFAULT_CLOUD_MODE = "false";
+  public static final String DEFAULT_REMOTE_ZOOKEEPER_HOST = "localhost:2181";
+  public static final String DEFAULT_REMOTE_SOLR_HOST = "localhost:8983";
+  private CloudSolrClient cloudClient;
+  private HttpSolrClient httpClient;
+
+  private Boolean cloudMode;
+
+  /**
+   * Initialize any state for this DB. Called once per DB instance; there is one DB instance per
+   * client thread.
+   */
+  @Override
+  public void init() throws DBException {
+    // initialize Solr driver
+    Properties props = getProperties();
+    String collectionName = props.getProperty("table", DEFAULT_COLLECTION_NAME);
+    // TO-DO: Add a check that the collection exists
+    // Check if Solr cluster is running in SolrCloud mode
+    cloudMode = Boolean.parseBoolean(props.getProperty("solr.cloud", DEFAULT_CLOUD_MODE));
+    System.out.println("Solr Cloud Mode = " + cloudMode);
+    if (cloudMode) {
+      System.out.println("Solr Zookeeper Remote Hosts = "
+          + props.getProperty("zookeeper.host.list", DEFAULT_REMOTE_ZOOKEEPER_HOST));
+
+      CloudSolrClient server = new CloudSolrClient(
+          props.getProperty("zookeeper.host.list", DEFAULT_REMOTE_ZOOKEEPER_HOST));
+      server.setDefaultCollection(collectionName);
+      cloudClient = server;
+    } else { // Start node only if cluster mode is disabled
+      HttpSolrClient server = new HttpSolrClient(
+          "http://" + props.getProperty("solr.host.list", DEFAULT_REMOTE_SOLR_HOST) + "/solr");
+      httpClient = server;
+    }
+  }
+
+  @Override
+  public void cleanup() throws DBException {
+    try {
+      if (cloudMode) {
+        cloudClient.close();
+      } else {
+        httpClient.close();
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Insert a record in the database. Any field/value pairs in the specified values HashMap will be
+   * written into the record with the specified record key.
+   *
+   * @param table
+   *          The name of the table
+   * @param key
+   *          The record key of the record to insert.
+   * @param values
+   *          A HashMap of field/value pairs to insert in the record
+   * @return Zero on success, a non-zero error code on error. See this class's description for a
+   *         discussion of error codes.
+   */
+  @Override
+  public Status insert(String table, String key, HashMap<String, ByteIterator> values) {
+    try {
+      SolrInputDocument doc = new SolrInputDocument();
+
+      doc.addField("id", key);
+      for (Entry<String, String> entry : StringByteIterator.getStringMap(values).entrySet()) {
+        doc.addField(entry.getKey(), entry.getValue());
+      }
+
+      if (cloudMode) {
+        cloudClient.add(table, doc);
+        cloudClient.commit(table);
```

Comment:
Explicit commits for each record will result in bad performance. Typically commits should be done within a certain amount of time or have the Solr server do auto soft/hard commits. This is different than a standard DB. It should be made configurable?